Formula/transmission-cli.rb

```diff
@@ -24,31 +20,45 @@ class TransmissionCli < Formula
     sha256 x86_64_linux:   "178d05964e9efd8d4541cf5589f4772ccdc59b8de83158f96b8ad7ffeff6b8d2"
   end
 
+  depends_on "cmake" => :build
+  depends_on "gettext" => :build
+  depends_on "ninja" => :build
   depends_on "pkg-config" => :build
   depends_on "libevent"
   depends_on "openssl@1.1"
 
   uses_from_macos "curl"
   uses_from_macos "zlib"
 
+  on_linux do
+    depends_on "python@3.11" => :build
+  end
+
   def install
+    args = %W[
+      -DCMAKE_INSTALL_PREFIX=pfx
+      -DCMAKE_BUILD_TYPE=RelWithDebInfo
+      -DCMAKE_INSTALL_PREFIX=#{prefix}
+      -DENABLE_DAEMON=ON
+      -DENABLE_QT=OFF
+      -DENABLE_WEB=OFF
+      -DENABLE_UTILS=ON
+      -DENABLE_CLI=ON
+      -DENABLE_MAC=OFF
+      -DENABLE_TESTS=OFF
+      -DENABLE_NLS=OFF
+    ]
+
     if OS.mac?
       ENV.append "LDFLAGS", "-framework Foundation -prebind"
       ENV.append "LDFLAGS", "-liconv"
+      args << "-DCMAKE_OSX_SYSROOT=#{MacOS.sdk_path}"
     end
 
-    args = %W[
-      --disable-dependency-tracking
-      --prefix=#{prefix}
-      --disable-mac
-      --disable-nls
-      --without-gtk
-    ]
-
-    system "./configure", *args
-    system "make", "install"
+    system "cmake", "-S", ".", "-B", "obj", "-G", "Ninja", *args
+    system "cmake", "--build", "obj", "--config", "RelWithDebInfo", "--target", "install/strip"
 
-    (var/"transmission").mkpath
+    (var/"pfx"/"bin")
```

Comment:
What should this line do?

Reply:
I’m not sure - I was assuming you needed to return the bin directory from this function?

Reply:
Which is what I thought the line I replaced did, for some strange reason! But we can remove this

Reply:
Not needed. Previously this `.mkpath` line was making a directory in specified path. Is it not needed anymore?